### PR TITLE
Time in sorting function

### PR DESCRIPTION
**Pull request by clementpoiret**
A not-yet-ready PoC to add time into the scoring function:

- Defines a T and R keybinding to set an estimated completion time (ect), or for example, 1d, 2h, 0.5w.
- Sorts overdue task first by applying an offset to it,
- Defines commands such as:
    - `:Dooing list`: displays the list of tasks and their indices,
    - `:Dooing set 1 priority important,urgent` to set the priorities of the 1st task,
    - `:Dooing set 1 ect 1d`, sets an estimated completion time of 1 day for the first task.

I didn't thoroughly tested the features, so this is only a draft :) 

Feel free to give me advices / feedbacks / suggestions !

### CODE REVIEW

**Comment by atiladefreitas** [comment]
I like the concept and the day/week logic.

think that colors can be same of prioritization. Here, both todos are checked as important.

image

and `.0` can be removed just for conciseness

**Comment by clementpoiret** [comment]
Thanks for your feedback, I'll change that. Also, the sorting of overdue tasks can be improved, I'll think about it 👌

**Comment by clementpoiret** [comment]
Okay, here is a stable version of the feature:

- Highlights are now only bound to priorities (we define a priority group in the config to attribute colors to it),
- In the json file, I renamed `priority` to `priorities` for consistency with the rest of the code,
- I refactored the logic behind the highlighting to make things a bit easier to read / understand,
- Tasks are sorted by the distance to the due date (e.g. nearest due date first),
- Tasks are can be done quickly must be done first. The current default config (which can be tweaked) forces long tasks (e.g. more than 12h to complete) to be penalized, and very quick tasks to be sorted first. Tasks that are neither important not urgent stay at the bottom.

2024-12-02-At-11h57m01s

What do you think? :)

**Comment by atiladefreitas** [comment]
LOVE THAT! gonna review and test locally later today

**Comment by atiladefreitas** [comment]
some conflicts is happening with the latest updates. Gonna check this out

**Comment by clementpoiret** [comment]
I solved the conflicts, let me know if it's good for you :)

File: README.md [hunk 1]
```diff
@@ -52,61 +52,73 @@ Dooing comes with sensible defaults that you can override:
...
-		padding = {
-			top = 1,
-			bottom = 1,
-			left = 2,
-			right = 2,
-		},
-	},
-
-    -- To-do formatting
-	formatting = {
```
Comment (atiladefreitas) [inline README.md:67]: this configuraton can´t be removed, added in the latest update

Comment (clementpoiret) [reply README.md:67]: Oops, sorry for the mistake, I just re-added it!

**Review by atiladefreitas — APPROVED** [review]

**Pull request merged**